### DISCUSSION TITLE
interact with wb metadata simply

### DISCUSSION THREAD

**Discussion by olivroy** [JanMarvin/openxlsx2#745] [Ideas]
Could this be done?
edit/access subject, title, keywords to an existing workbook. 

follow-up to discussion in #741

**Comment by JanMarvin** [comment]
Please don't open three liner issues for discussion topics. It could be done, but what about the current implementation, how about you open a pull request and have a look. It currently behaves the way it does, because nobody cared so far 😉 

That said, I currently don't plan to work on this, I have my hands full with release testing, xlsb and apparently flextable support...

Just add it to the load script and if you want to modify stuff add that too. The non trivial part is to not delay the load part to much and to not brick workbooks with broken XML

**Reply by JanMarvin** [reply]
Currently we use this to create a core file afaik only if none is present. When loading the file. `wb$subject` is only a relict without real use.

https://github.com/JanMarvin/openxlsx2/blob/1f43b558e545ce31ce4ae66d5b2400cbca196075/R/baseXML.R#L155 

We could replace this with a `wb_get_properties()`/`wb_set_properties()`. Internally we should keep it XML so that we don't have to deal with rare cases for properties seldomly used.

**Reply by olivroy** [reply]
Thanks. I wil try to have a look, but not convinced I will accomplish something meaningful :) I am scared to break things

**Reply by JanMarvin** [reply]
Have a look at `L.6.2 Metadata` in `Ecma Office Open XML Part 1 - Fundamentals And Markup Language Reference.pdf` (p. 5011) for some documentation. Otherwise I'd have to research how to break things :)

**Comment by olivroy** [comment]
Done with `wb_get/set_properties()` https://janmarvin.github.io/openxlsx2/dev/reference/properties-wb.html
